File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Effect/Combat/WebEffect.ts
import { TimeKit } from "../../../../../Core/Utils/TimeKit";
import DungeonManager from "../../DungeonManager";
import AbstractCreature from "../../Unit/AbstractCreature";
import { AbstractEffect } from "../AbstractEffect";
import { WebLineEffect } from "./WebLineEffect";
import { WebParticleEffect } from "./WebParticleEffect";


export class WebEffect extends AbstractEffect{
    private _Timer = 0.1;
    private _Count = 0;
    public constructor(private _Target:AbstractCreature,private _X:number,private _Y:number) {
        super();
    }
    
    public Update() {
        this._Timer -= TimeKit.DeltaTime;
        if(this._Timer < 0){
            this._Timer += 0.1;
            switch(this._Count){
                case 0:{
                    DungeonManager.EffectManager.Play(new WebLineEffect(this._X,this._Y,true));
                    DungeonManager.EffectManager.Play(new WebLineEffect(this._X,this._Y,true));
                    DungeonManager.EffectManager.Play(new WebParticleEffect(this._X - 90,this._Y - 10));
                    break;
                }
                case 1:{
                    DungeonManager.EffectManager.Play(new WebLineEffect(this._X,this._Y,true));
                    DungeonManager.EffectManager.Play(new WebLineEffect(this._X,this._Y,true));
                    break;
                }
                case 2:{
                    DungeonManager.EffectManager.Play(new WebLineEffect(this._X,this._Y,true));
                    DungeonManager.EffectManager.Play(new WebLineEffect(this._X,this._Y,true));
                    DungeonManager.EffectManager.Play(new WebParticleEffect(this._X + 70,this._Y + 80));
                    break;
                }
                case 4:{
                    DungeonManager.EffectManager.Play(new WebParticleEffect(this._X + 30,this._Y - 100));
                    break;
                }
            }
            this._Count++;
        }
        if(this.Duration >= 1) {
            this.IsDone = true;
            return;
        }
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Combat/WebEffect.cs
using ModTheSpire.Dungeon;
using ModTheSpire.Unit;
using UnityEngine;

namespace ModTheSpire.Effect.Combat
{
    public class WebEffect:AbstractEffect
    {
        private AbstractCreature mTarget;
        private float mX;
        private float mY;
        private float mTimer;
        private int mCount;

        public WebEffect(AbstractCreature target, float x, float y)
        {
            mTarget = target;
            mX = x;
            mY = y;
        }

        public override void Update()
        {
            base.Update();
            this.mTimer -= Time.deltaTime;
            if(this.mTimer < 0){
                this.mTimer += 0.1f;
                var obj = this.mTarget.DisplayObject.GetHitBox();
                switch(this.mCount){
                    case 0:{
                        DungeonManager.EffectManager.Play(new WebLineEffect(mX,mY,true));
                        DungeonManager.EffectManager.Play(new WebLineEffect(mX,mY,true));
                        DungeonManager.EffectManager.Play(new WebParticleEffect(obj.center.x - 90,obj.center.y - 10));
                        break;
                    }
                    case 1:{
                        DungeonManager.EffectManager.Play(new WebLineEffect(mX,mY,true));
                        DungeonManager.EffectManager.Play(new WebLineEffect(mX,mY,true));
                        break;
                    }
                    case 2:{
                        DungeonManager.EffectManager.Play(new WebLineEffect(mX,mY,true));
                        DungeonManager.EffectManager.Play(new WebLineEffect(mX,mY,true));
                        DungeonManager.EffectManager.Play(new WebParticleEffect(obj.center.x + 70,obj.center.y + 80));
                        break;
                    }
                    case 4:{
                        DungeonManager.EffectManager.Play(new WebParticleEffect(obj.center.x + 30,obj.center.y - 100));
                        break;
                    }
                }
                this.mCount++;
            }
            if(Duration >= 1)
                Finish();
        }
    }
}